### PR TITLE
Df use bparser

### PR DESCRIPTION
**Pull request by dflanderka**
- add BParser as submodule
- use in FieldFormula

### CODE REVIEW

File: unit_tests/fields/bparser_test_tools.hh [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/*
```
Comment (jbrezmorf) [inline unit_tests/fields/bparser_test_tools.hh]: No need to having copy of this file, just put some functions into bparser test.

**Comment by jbrezmorf** [comment]
Update in BParser (commit 
8701091).  Basic speed test gives performance about one-third of the native C code. A lot of bug fixes as well.

File: CMakeLists.txt [hunk 2]
```diff
@@ -205,7 +205,7 @@ if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
       # not cygwin 
       # is version 4.7 or higher?
       elseif (GCC_VERSION VERSION_GREATER 4.7 OR GCC_VERSION VERSION_EQUAL 4.7)
-          set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+          set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fext-numeric-literals")
```
Comment (jbrezmorf) [inline CMakeLists.txt]: Why the new option. Seems it is irrelevant for c++11 and later.

File: src/fields/field_value_cache.hh [hunk 3]
```diff
@@ -43,7 +43,7 @@ template<class elm_type>
 class FieldValueCache {
 public:
     /// Constructor
-    FieldValueCache(unsigned int n_rows, unsigned int n_cols);
+    FieldValueCache(unsigned int n_rows = 1, unsigned int n_cols = 1);
```
Comment (jbrezmorf) [inline src/fields/field_value_cache.hh:46]: Jelikoz je cache pouzivana interne algoritmy pro assemblaci asi bude lepsi specifikovat ty parametry vzdy axplicitne.

File: src/fields/field_formula.cc [hunk 4]
```diff
@@ -231,6 +242,83 @@ void FieldFormula<spacedim, Value>::value_list (const Armor::array &point_list,
...
+                auto cache_val = data_cache.data().template mat<Value::NRows_, Value::NCols_>(i);
+                cache_val(row, col) = res_[i];
+                data_cache.data().set(i) = cache_val;
+            }
+        }
+}
+
+
+template <int spacedim, class Value>
+void FieldFormula<spacedim, Value>::cache_reinit(const ElementCacheMap &cache_map)
```
[the remark this answers is not in the document]
Comment (dflanderka) [reply src/fields/field_formula.cc:287]: Described to comment in #1235

**Pull request merged**